File: Library.py
from Book import book
class library:
    def __init__(self):
        self.users = []
        self.books = []

    def printUsers(self):
        i = 1
        has_rented = ""
        if not self.users:
            print("No hay usuarios registrados\n")
            return
        print("\t=====LISTA DE USUARIOS======")
        for user in self.users:
            has_rented = "Si" if user.hasRented else "No"
            print(f"ID: {i} | Nombre: {user.name} {user.lastName} | Ha rentado: {has_rented}")
            i += 1
        print()

    def printBooks(self):
        i = 1
        is_rented = ""
        if not self.books:
            print("No hay libros registrados\n")
            return
        print("\t=====LISTA DE LIBROS======")
        for book in self.books:
            is_rented = "Si" if book.isRented else "No"
            print(f"{i}. Libro: {book.name} | Autor: {book.author} | Esta rentado: {is_rented}")
            i += 1
        print()

    def printRentedBooks(self):
        i = 1
        if not self.books:
            print("No hay libros registrados\n")
            return
        print("\t=====LISTA DE LIBROS RENTADOS======")
        for book in self.books:
            if book.isRented:
                print(f"{i}. Libro: {book.name} | Autor: {book.author} | Lo rento: {book.user}")
                i += 1
        print()

    def printUsersWithBooks(self):
        i = 1
        if not self.users:
            print("No hay usuarios registrados\n")
            return
        print("\t=====LISTA DE USUARIOS QUE HAN RENTADO======")
        for user in self.users:
            if user.hasRented:
                print(f"{i}. Nombre: {user.name} {user.lastName}")
                print(f"\tLibros que ha rentado {user.name}: ")
                for book in user.booksRented:
                    print(f"\t\tLibro: {book.name} | Autor: {book.author}")
                i += 1
        print()

File: Library.java
import java.util.*;
public class Library {
    private ArrayList<User> users = new ArrayList<User>();
    private ArrayList<Book> books = new ArrayList<Book>();

    public void printUsers(){
        int i = 1;
        String hasRented = "";
        if (users.isEmpty()){
            System.out.println("No hay usuarios registrados\n");
            return;
        }
        System.out.println("\t=====LISTA DE USUARIOS======");
        for (User usuarios: this.users){
            if (usuarios.isHasRented()) {
                hasRented = "Si";
            } else {
                hasRented = "No";
            }
            System.out.println("ID: "+i+" | Nombre: "+usuarios.getName()+" "+usuarios.getLastName()+" | Ha rentado: "+hasRented);
            i++;
        }
        System.out.println();
    }

    public void printBooks(){
        int i = 1;
        String isRented = "";
        if (books.isEmpty()){
            System.out.println("No hay libros registrados\n");
            return;
        }
        System.out.println("\t=====LISTA DE LIBROS======");
        for (Book libros: this.books){
            if (libros.isRented()) {
                isRented = "Si";
            } else {
                isRented = "No";
            }
            System.out.println(i+". Libro: "+libros.getName()+" | Autor: "+libros.getAuthor()+" | Esta rentado: "+isRented);
            i++;
        }
        System.out.println();
    }

    public void printRentedBooks(){
        int i = 1;
        int c = 0;
        for (Book libros: this.books){
            if (libros.isRented()) {
                c++;
            }
        }
        if (c == 0) {
            System.out.println("No hay libros rentados\n");
            return;
        }
        System.out.println("\t=====LISTA DE LIBROS RENTADOS======");
        for (Book libros: this.books){
            if (libros.isRented()) {
                System.out.println(i + ". Libro: " + libros.getName() + " | Autor: " + libros.getAuthor()+" | Lo rento: "+libros.getUser());
                i++;
            }
        }
        System.out.println();
    }
    public void printUsersWithBooks(){
        int i = 1;
        int c = 0;
        for (User usuarios: this.users){
            if (usuarios.isHasRented()) {
                c++;
            }
        }
        if (c == 0) {
            System.out.println("No hay usuarios con libros\n");
            return;
        }
        System.out.println("\t=====LISTA DE USUARIOS QUE HAN RENTADO======");
        for (User usuarios: this.users){
            if (usuarios.isHasRented()) {
                System.out.println(i + ". Nombre: " + usuarios.getName() + " " + usuarios.getLastName());
                System.out.println("\tLibros que ha rentado "+usuarios.getName()+": ");
                for (Book libros : usuarios.getBooksRented()){
                    System.out.println("\t\tLibro: " + libros.getName() + " | Autor: " + libros.getAuthor());
                }
                i++;
            }
        }
        System.out.println();
    }

    public ArrayList<User> getUsers() {
        return users;
    }

    public ArrayList<Book> getBooks() {
        return books;
    }
}

